libs/dal/src/repositories/channel-endpoint/channel-endpoint.schema.ts

```diff
@@ -0,0 +1,68 @@
+import { ChannelEndpointRoutingTypeEnum } from '@novu/shared';
+import mongoose, { Schema } from 'mongoose';
+import { schemaOptions } from '../schema-default.options';
+import { ChannelEndpointDBModel } from './channel-endpoint.entity';
+
+const routingSchema = new Schema(
+  {
+    type: {
+      type: String,
+      required: true,
+      enum: Object.values(ChannelEndpointRoutingTypeEnum),
+    },
+  },
+  {
+    _id: false,
+    discriminatorKey: 'type',
+  }
+);
+
+// Slack routing schema
+const slackRoutingSchema = new Schema(
+  {
+    channelId: {
+      type: String,
+      required: false,
+    },
+    userId: {
+      type: String,
+      required: false,
+    },
+  },
+  { _id: false }
+);
+
+routingSchema.discriminator(ChannelEndpointRoutingTypeEnum.SLACK, slackRoutingSchema);
+
+const channelEndpointSchema = new Schema<ChannelEndpointDBModel>(
+  {
+    identifier: {
```

Comment:
What this will be used for?

Reply:
Its a unique identifier (unique per environment) for the CRUD operations like we do in the other endpoints. We could expose just MongoDB but I think identifier makes more sense.